### PR TITLE
MULE-11161: change PGP encryption cipher from CAST5 to AES-256

### PR DESCRIPTION
**Pull request by alepulver**

### CODE REVIEW

**Comment by marianogonzalez** [comment]
this should include a note in the migration guide. Also, is there a use case in which the user might one to go back to the prior one? Shouldn't we have a mechanism to specify the cipher?

**Comment by aiannucci** [comment]
I agree that this should be user configurable. Also, I'm almost sure that this change breaks the security module implementation. Changes will be needed there as well.

**Comment by alepulver** [comment]
Yes, I'll add the note. We could add a system property, but it wasn't configurable before.

**Comment by marianogonzalez** [comment]
@alepulver I understand that it wasn't configurable before, but you cannot claim backwards compatibility without a mechanism to go back to the old behavior.

As @aiannucci said:

* you need to check how this impacts AES. If a new AES version is needed to work with this change, then that has to be clearly documented on the migration guide and you need to let the documentation team now as well.
* A system property is a good way to do it if this fix needs to go on a patch release. However, if this is only for 3.9 onwards, then make it explicit on the DSL.

**Comment by marianogonzalez** [comment]
where does the need for this change comes from anyway? Is this source of problems or did you just saw an easy improvement and went for it?

**Comment by alepulver** [comment]
@marianogonzalez the issue was assigned to me with high priority. I'll make it configurable for future releases and make a patch for the customer that requested it.

File: modules/pgp/src/main/java/org/mule/module/pgp/KeyBasedEncryptionStrategy.java [hunk 1]
```diff
@@ -43,6 +45,11 @@ public void initialise() throws InitialisationException
             java.security.Security.addProvider(new BouncyCastleProvider());
         }
         provider = SecurityUtils.getDefaultSecurityProvider();
+
+        if (encryptionAlgorithm == null)
+        {
+            encryptionAlgorithm = SymmetricKeyAlgorithmTags.CAST5;
```
Comment (marianogonzalez) [inline modules/pgp/src/main/java/org/mule/module/pgp/KeyBasedEncryptionStrategy.java]: static import

File: modules/pgp/src/main/java/org/mule/module/pgp/KeyBasedEncryptionStrategy.java [hunk 2]
```diff
@@ -137,4 +144,9 @@ public void setCheckKeyExpirity(boolean checkKeyExpirity)
     {
         this.checkKeyExpirity = checkKeyExpirity;
     }
+
+    public void setEncryptionAlgorithm(Integer encryptionAlgorithm)
+    {
+        this.encryptionAlgorithm = encryptionAlgorithm;
```
Comment (marianogonzalez) [inline modules/pgp/src/main/java/org/mule/module/pgp/KeyBasedEncryptionStrategy.java]: where is this setter invoked? I don't see any code using it

Comment (alepulver) [inline modules/pgp/src/main/java/org/mule/module/pgp/KeyBasedEncryptionStrategy.java]: From the seccurity module.

**Pull request merged**